### DISCUSSION TITLE
Why is EventId with ID set to 0 not being added to OTLP Log Record?

### DISCUSSION THREAD

**Discussion by davidkaya** [open-telemetry/opentelemetry-dotnet#4779] [Q&A] [answered]
Hello,

I have noticed, that if the log record has an `EventId` with `ID` set to 0, the `EventId` will no be added to the OTLP Log Record: https://github.com/open-telemetry/opentelemetry-dotnet/blob/5784b45c4e310fc54d84278891bb22d6b93a1246/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs#L96-L99

Is there some specific reason behind that? Is this possibly a bug?

Thank you

**Comment by cijothomas** [comment, marked as answer]
0 is the default, i.e its treated as if eventid is not set at all. 

Separately, EventId will be temporarily removed from OTLP Exporter to stabilize it. See https://github.com/open-telemetry/opentelemetry-dotnet/issues/4776

**Reply by davidkaya** [reply]
That's unfortunate though. Even ASP NET Core itself has few logs with EventId set to 0. It might be nice to revisit the "default" event id when it will be stabilized.

**Reply by cijothomas** [reply]
Can you point to the places where Asp.Net Core uses EventId 0? We can ask the owners to see if that was intentional..

**Reply by davidkaya** [reply]
I have reported an issue here https://github.com/dotnet/runtime/issues/90831
It contains links to ASP.NET Core repo
